mautic/mautic
# Stream contact exports

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | ✔️ 
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

Exporting from the contacts page used to load all contacts in the memory, thus resulting in a timeout or an out of memory error. This PR addresses this issue by streaming the contact exports.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Export over 10K contacts from the contacts page using a filter
2. The export should probably crash or timeout

#### Steps to test this PR:
1. Apply this PR
1. Export the same contacts
2. The export should be streamed incrementally :tada:
